ValidationMixin.js

```diff
@@ -44,18 +43,40 @@ var ValidationMixin = {
    * Convenience method to validate a key via an event handler. Useful for
    * onBlur, onClick, onChange, etc...
    *
-   * @param {?String} State key to validate
+   * Usage: <input onBlur={this.handleUnfocusFor('password')} .../>
+   *
+   * @param {?String} key to validate
    * default is false.
    * @return {function} validation event handler
    */
-  handleValidation: function(key) {
-    return function(event) {
+  handleUnfocusFor: function(key) {
+    return function handleUnfocus(event) {
       event.preventDefault();
       this.validate(key);
     }.bind(this);
   },
 
   /**
+   * Convenience method to validate a whole form on submit
+   *
+   * Usage: <form onSubmit={this.handleSubmit}>...</form>
+   */
+  handleSubmit: function(event) {
```

Comment:
I'm not sure about this from a developers perspective, we cannot generically handle the submit without providing a hook or callback to execute if validation is successful. I have an idea about how to handle this but am still working it out. I'll push it soon at let you review.